### PR TITLE
Ensures split between hosted/self-hosted integrations

### PR DESCRIPTION
**Pull request by eamodio**
Fixes possible auth issues with self-hosted integrations

### CODE REVIEW

File: src/plus/integrations/providers/gitlab.ts [hunk 1]
```diff
@@ -157,19 +151,31 @@ export class GitLabIntegration extends ProviderIntegration<GitLabRepositoryDescr
...
+	}
+
+	protected get apiBaseUrl(): string {
+		return 'https://gitlab.com/api';
+	}
+}
+
+export class GitLabSelfHostedIntegration extends GitLabIntegrationBase<SelfHostedProviderId.GitHubEnterprise> {
+	readonly authProvider = enterpriseAuthProvider;
+	readonly id = SelfHostedProviderId.GitHubEnterprise;
```
Comment (axosoft-ramint) [inline src/plus/integrations/providers/gitlab.ts:170]: This should be `GitLabEnterprise`

File: src/plus/integrations/providerIntegration.ts [hunk 2]
```diff
@@ -26,22 +26,30 @@ import type {
...
+export type ProviderKeyById<T extends SupportedProviderIds> = T extends SupportedHostedProviderIds
+	? `${SupportedHostedProviderIds}`
+	: `${SupportedSelfHostedProviderIds}:${string}`;
 
 export type RepositoryDescriptor = { key: string } & Record<string, unknown>;
 
-export abstract class ProviderIntegration<T extends RepositoryDescriptor = RepositoryDescriptor> {
+export abstract class ProviderIntegration<
+	ID extends SupportedProviderIds = SupportedProviderIds,
+	T extends RepositoryDescriptor = RepositoryDescriptor,
```
Comment (axosoft-ramint) [inline src/plus/integrations/providerIntegration.ts:51]: While we are refactoring all this, can we do away with requiring a repository descriptor for an integration? Certain integrations (like Jira, which I'm looking at now) don't support repositories - they're just issue providers that work with boards or projects.

**Review by axosoft-ramint — COMMENTED** [review]
Some preliminary feedback on a quick scan before I start testing

File: src/plus/integrations/providers/gitlab.ts (hunk 1 above)
Comment (eamodio) [reply src/plus/integrations/providers/gitlab.ts:170]: Please feel free to commit and take over this.

File: src/plus/integrations/providerIntegration.ts (hunk 2 above)
Comment (eamodio) [reply src/plus/integrations/providerIntegration.ts:51]: I'm not sure the implications of removing that yet (haven't had the time to look), but I would suggest doing that outside of this.

Comment (axosoft-ramint) [reply src/plus/integrations/providerIntegration.ts:51]: Ok, no worries, I can include that in the Jira stuff.

File: src/plus/integrations/providers/gitlab.ts (hunk 1 above)
Comment (axosoft-ramint) [reply src/plus/integrations/providers/gitlab.ts:170]: Roger that 👍🏼

**Review by axosoft-ramint — APPROVED** [review]
I don't have a GitLab Self-Hosted instance to test with, but given that it's currently broken, and given that this is correcting functions like `getByRemote` where the self-hosted context was previously being lost, I'd say it's safe to merge.

I did manage to test the Focus View since some updated functions touch that - it works fine 👍🏼

**Pull request merged**